pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -345,23 +345,29 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     if (offlineTableName != null && realtimeTableName != null) {
       // Hybrid
       offlineBrokerRequest = getOfflineBrokerRequest(brokerRequest);
-      _queryOptimizer.optimize(offlineBrokerRequest.getPinotQuery(), schema);
+      _queryOptimizer.optimize(offlineBrokerRequest.getPinotQuery(),
+          _tableCache.getTableConfig(TableNameBuilder.OFFLINE.tableNameWithType(rawTableName)), schema);
```

Comment:
Same for other places
```suggestion
      _queryOptimizer.optimize(offlineBrokerRequest.getPinotQuery(), _tableCache.getTableConfig(offlineTableName), schema);
```

Reply:
Fixed.